### PR TITLE
[Distributed] Introduce "roundtrip" test and AS, remove duplication and cleanups

### PR DESCRIPTION
**Pull request by ktoso**
Followup to introduction of remote calls.

Introduces "roundtrip" tests and does a lot of cleanups.

- move func "getters" to ASTContext
- introduce reusable "roundtrip actor system"
- introduce "roundtrip" test

Will add more commit by commit.

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test and merge

File: test/Distributed/Inputs/FakeDistributedActorSystems.swift [hunk 1]
```diff
@@ -126,3 +126,238 @@ public struct FakeInvocation: DistributedTargetInvocationEncoder, DistributedTar
...
+        self.remoteCallResult = value
+        self.remoteCallError = nil
+      } onError: { error in
+        self.remoteCallResult = nil
+        self.remoteCallError = error
+      }
+      try await executeDistributedTarget(
+        on: active,
+        mangledTargetName: target.mangledName,
+        invocationDecoder: &invocation,
```
Comment (xedin) [inline test/Distributed/Inputs/FakeDistributedActorSystems.swift:205]: FYI - I'm dropping `inout` in my upcoming PR.

File: test/Distributed/Inputs/FakeDistributedActorSystems.swift [hunk 2]
```diff
@@ -126,3 +126,238 @@ public struct FakeInvocation: DistributedTargetInvocationEncoder, DistributedTar
...
+      default:
+        fatalError("No reply!")
+      }
+    }
+    try await _openExistential(targetActor, do: doIt)
+  }
+
+}
+
+public struct FakeRoundtripInvocation: DistributedTargetInvocationEncoder, DistributedTargetInvocationDecoder {
```
Comment (xedin) [inline test/Distributed/Inputs/FakeDistributedActorSystems.swift:271]: This is going to be a class.

**Review by xedin — COMMENTED** [review]
LGTM!

**Pull request merged**